pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/ControllerVipBasedSegmentUploader.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.manager.realtime;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.apache.pinot.common.metrics.ServerMetrics;
+import org.apache.pinot.common.protocols.SegmentCompletionProtocol;
+import org.apache.pinot.common.utils.FileUploadDownloadClient;
+import org.apache.pinot.core.util.SegmentCompletionProtocolUtils;
+import org.slf4j.Logger;
+
+import static org.apache.pinot.common.protocols.SegmentCompletionProtocol.ControllerResponseStatus.UPLOAD_SUCCESS;
+
+
+// A segment uploader which uploads segments to the controller via the controller's segmentCommitUpload end point
+// point.
+public class ControllerVipBasedSegmentUploader implements SegmentUploader {
```

Comment:
Maybe call it `Server2ControllerSegmentUploader`? I think we should omit Vip in the name, there is nothing in here that says we should go through vip

Reply:
Renamed.